walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -28,247 +28,356 @@
+        //
+        // Events  : |--A--|     |----A----|
+        // Day     : |---------------------|
+        // Options :       |-----|
+
+        Collection<Event> events = Arrays.asList(
+            new Event("Event 1", TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0830AM, false),
+                Arrays.asList(PERSON_A)),
+            new Event("Event 2", TimeRange.fromStartEnd(TIME_0900AM, TimeRange.END_OF_DAY, true),
+                Arrays.asList(PERSON_A)));
+
+        MeetingRequest request = new MeetingRequest(Arrays.asList(PERSON_A), DURATION_30_MINUTES);
+
+        Collection<TimeRange> actual = query.query(events, request);
+        Collection<TimeRange> expected =
+            Arrays.asList(TimeRange.fromStartDuration(TIME_0830AM, DURATION_30_MINUTES));
+
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void ignoresPeopleNotAttending() {
+        // Add an event, but make the only attendee someone different from the person looking to book
+        // a meeting. This event should not affect the booking.
+        Collection<Event> events = Arrays.asList(new Event("Event 1",
+            TimeRange.fromStartDuration(TIME_0900AM, DURATION_30_MINUTES), Arrays.asList(PERSON_A)));
+        MeetingRequest request = new MeetingRequest(Arrays.asList(PERSON_B), DURATION_30_MINUTES);
+
+        Collection<TimeRange> actual = query.query(events, request);
+        Collection<TimeRange> expected = Arrays.asList(TimeRange.WHOLE_DAY);
+
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void noConflicts() {
+        MeetingRequest request =
+            new MeetingRequest(Arrays.asList(PERSON_A, PERSON_B), DURATION_30_MINUTES);
+
+        Collection<TimeRange> actual = query.query(NO_EVENTS, request);
+        Collection<TimeRange> expected = Arrays.asList(TimeRange.WHOLE_DAY);
+
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void notEnoughRoom() { 
+        // Have one person, but make it so that there is not enough room at any point in the day to
+        // have the meeting.
+        //
+        // Events  : |--A-----| |-----A----|
+        // Day     : |---------------------|
+        // Options :
+
+        Collection<Event> events = Arrays.asList(
+            new Event("Event 1", TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0830AM, false),
+                Arrays.asList(PERSON_A)),
+            new Event("Event 2", TimeRange.fromStartEnd(TIME_0900AM, TimeRange.END_OF_DAY, true),
+                Arrays.asList(PERSON_A)));
+
+        MeetingRequest request = new MeetingRequest(Arrays.asList(PERSON_A), DURATION_60_MINUTES);
+
+        Collection<TimeRange> actual = query.query(events, request);
+        Collection<TimeRange> expected = Arrays.asList();
+
+        Assert.assertEquals(expected, actual);
+    }
+
+
+    // UNIT TESTS THAT I WROTE.
+
+
+    @Test
+    public void optionalAttendeeWithAllDayMeeting() { 
```

Comment:
The comments indicating what the events looked like were really helpful. Maybe add those here? If it's too much work you can ignore.